### PR TITLE
Updated basket creation logic to use site

### PR DESCRIPTION
**Pull request by clintonb**
This ensures baskets are tied to a single site/tenant, and we avoid leakage across sites.

ECOM-2583

### CODE REVIEW

**Comment by clintonb** [comment]
@cpennington @rlucioni please review.

File: ecommerce/extensions/partner/shortcuts.py [hunk 1]
```diff
@@ -1,12 +1,3 @@
...
 def get_partner_for_site(request):
-    """Get the partner for the requested site"""
-    try:
-        site = get_current_site(request)
-    except Site.DoesNotExist:
-        return None
-
-    return site.siteconfiguration.partner
+    """ Returns the Partner associated with the request. """
+    return request.site.siteconfiguration.partner
```
Comment (cpennington) [inline ecommerce/extensions/partner/shortcuts.py:3]: Does `request` always have a `site`? If so, why were we using `get_current_site` before?

Comment (clintonb) [inline ecommerce/extensions/partner/shortcuts.py:3]: I added the [`site middleware`](https://docs.djangoproject.com/en/1.8/ref/contrib/sites/#site-middleware). The previous developer was probably unaware of the middleware.

**Comment by clintonb** [comment]
Merging so that we can keep this project moving, and deploy today.

**Pull request merged**